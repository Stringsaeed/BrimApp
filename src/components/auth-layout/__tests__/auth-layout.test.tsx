/* eslint-disable @typescript-eslint/no-require-imports */
import { screen } from "@testing-library/react-native";
import React from "react";
import { Text } from "react-native";

import render from "utils/test";

import AuthLayout from "../";

jest.mock("react-native-reanimated", () => {
  const mockedReanimated = require("react-native-reanimated/mock");
  return {
    __esModule: true,
    ...mockedReanimated,
    useAnimatedKeyboard: jest.fn(() => {
      return {
        height: {
          value: 0,
        },
      };
    }),
  };
});

describe("AuthLayout", () => {
  it("should render correctly", () => {
    render(
      <AuthLayout>
        <Text>Test</Text>
      </AuthLayout>
    );

    expect(screen.toJSON()).toMatchSnapshot();
  });

  it("should render children", () => {
    render(
      <AuthLayout>
        <Text>Test</Text>
      </AuthLayout>
    );

    expect(screen.getByText("Test")).toBeTruthy();
  });
});
