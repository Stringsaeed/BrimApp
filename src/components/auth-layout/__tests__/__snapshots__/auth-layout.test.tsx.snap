// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthLayout should render correctly 1`] = `
<View
  offset={40}
  style={
    {
      "backgroundColor": "rgba(255,255,255,0)",
      "flex": 1,
      "flexDirection": "column",
      "gap": 24,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 24,
    }
  }
  testID="animated-keyboard-view"
>
  <Text>
    Test
  </Text>
</View>
`;
